internal/exp/metrics/metrics_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics_test
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+func TestMergeMetrics(t *testing.T) {
```

Comment:
Can you add a benchmark as well? Perhaps one comparing a plain copyto from B into A vs. the new Merge function in this PR.

Reply:
Done. Please double check my assumptions, but it looks to be similar performance (a copy is a copy), but depending on the situation, potentially drastically less memory allocations

```
Running tool: /usr/local/go/bin/go test -benchmem -run=^$ -bench . github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics

goos: linux
goarch: amd64
pkg: github.com/open-telemetry/opentelemetry-collector-contrib/internal/exp/metrics
cpu: Intel(R) Xeon(R) CPU E5-2687W v4 @ 3.00GHz
BenchmarkMergeManyIntoSingle/Naive-24         	     100	  14482976 ns/op	 5990548 B/op	  130017 allocs/op
BenchmarkMergeManyIntoSingle/Deduplicating-24 	      79	  16174993 ns/op	 1671475 B/op	   30018 allocs/op
BenchmarkMergeManyIntoMany/Naive-24           	      96	  12998815 ns/op	 6106008 B/op	  130003 allocs/op
BenchmarkMergeManyIntoMany/Deduplicating-24   	     100	  14233288 ns/op	 1672314 B/op	   30019 allocs/op
BenchmarkNaiveMerge-24                        	1000000000	         0.0000004 ns/op	       0 B/op	       0 allocs/op
```

Reply:
I could also potentially create a benchmark that compares serialization time with naive vs deduplicating merge